src/components/InfiniteScrollPaginator/InfiniteScroll.tsx

```diff
@@ -10,43 +12,78 @@ const mousewheelListener = (event: Event) => {
   }
 };
 
-export type InfiniteScrollProps = {
+export type InfiniteScrollProps = PaginatorProps & {
   className?: string;
   element?: React.ElementType;
+  /**
+   * @desc Flag signalling whether more pages with older items can be loaded
+   * @deprecated Use hasPreviousPage prop instead. Will be removed with v11.0.0.
+   */
   hasMore?: boolean;
+  /**
+   * @desc Flag signalling whether more pages with newer items can be loaded
+   * @deprecated Use hasNextPage prop instead. Will be removed with v11.0.0.
+   */
   hasMoreNewer?: boolean;
-  /** Element to be rendered at the top of the thread message list. By default Message and ThreadStart components */
+  /** Element to be rendered at the top of the thread message list. By default, Message and ThreadStart components */
   head?: React.ReactNode;
   initialLoad?: boolean;
   isLoading?: boolean;
   listenToScroll?: (offset: number, reverseOffset: number, threshold: number) => void;
   loader?: React.ReactNode;
-  loading?: React.ReactNode;
+  /**
+   * @desc Function that loads previous page with older items
+   * @deprecated Use loadPreviousPage prop instead. Will be removed with v11.0.0.
+   */
   loadMore?: () => void;
+  /**
+   * @desc Function that loads next page with newer items
+   * @deprecated Use loadNextPage prop instead. Will be removed with v11.0.0.
```

Comment:
I'm not sure whether we can plan the exact version where this property will get removed. I think we should start with creating a Github issue to remind us for cleaning this up and then we could plan/prioritize it accordingly.

Reply:
@oliverlaz I have create the issue and its link to the comments. Thank you for reminding me this.